src/veh_interact.cpp

```diff
@@ -934,79 +927,79 @@ void veh_interact::do_mend()
 
 }
 
-/**
- * Handles refilling a vehicle's fuel tank.
- * @param reason INVALID_TARGET if there's no fuel tank in the spot,
- *               CANT_REFILL All tanks are broken or player has nothing to fill the tank with.
- */
 void veh_interact::do_refill()
 {
-    const task_reason reason = cant_do('f');
-    display_mode('f');
-    werase (w_msg);
-    int msg_width = getmaxx(w_msg);
+    display_mode( 'f' );
+    werase( w_msg );
 
-    switch (reason) {
-    case INVALID_TARGET:
-        mvwprintz(w_msg, 0, 1, c_ltred, _("There's no refillable part here."));
-        wrefresh (w_msg);
-        return;
-    case CANT_REFILL:
-        mvwprintz(w_msg, 1, 1, c_ltred, _("There is no refillable part here."));
-        mvwprintz(w_msg, 2, 1, c_white, _("The part is broken, or you don't have the "
-                                          "proper fuel."));
-        wrefresh (w_msg);
-        return;
-    case MOVING_VEHICLE:
-        fold_and_print( w_msg, 0, 1, msg_width - 2, c_ltgray,
-                        _( "You can't refill the vehicle while driving." ) );
-        wrefresh (w_msg);
-        return;
-    default:
-        break; // no reason, all is well
+    // get all tanks and determine if possible to refill with currently available items
+    std::vector<std::pair<vehicle_part *, bool>> tanks;
+    for( auto &e : veh->parts ) {
+        if( e.is_tank() ) {
+            tanks.emplace_back( &e, std::any_of( parts_here.begin(), parts_here.end(), [&]( int idx ) {
+                auto func = [&]( const item &e ) {
```

Comment:
This `e` here is quite confusing. It hides the "outer" `e` and, given this, the whole `std::any_of( ... )` doesn't depend on the outer `e` at all. Which means you get the same `std::any_of()` as many times as you have parts in the vehicle. Was that intentional? 

Hidden iterators are much worse than `column` / `columns`.